clerk-frontend/src/need-wizard/ConfirmStep.jsx

```diff
@@ -9,43 +7,56 @@ const timeoutPromise = (timeoutMillis) => {
   })
 }
 
-class ConfirmStep extends Component {
-
+class ConfirmStep extends React.Component {
   constructor(props) {
     super(props);
     this.state = {}
-
     this.abundance = new AbundanceService(this.props.core)
   }
 
   async componentDidMount() {
     await timeoutPromise(100)
 
+    const siteDid = 'did:discipl:ipv8:TGliTmFDTFBLOpd7y5Yd9NDKxelDmsACx9I6Go2cK4v8fKBIKzp0KiEjStqcAfSHFfk7KkqbUodAH0iX3nljnd7Y+bj8b0oFUCI='
+    const serviceDid = 'did:discipl:ipv8:TGliTmFDTFBLOs9RF8NUdlFdHcJaSZlNH4F0vuwSB3epF8s8ns1NcB4fWcKSQcuWuqj3C/RQIk3fEEwSwpodkfNmJW54loFalTI='
+
     let need = await this.abundance.need('ephemeral', this.props.need)
+    let ipv8TempLink = await this.abundance.getCoreAPI().claim(
+      { did: siteDid },
+      this.props.need,
+      { did: serviceDid }
+    )
 
     let observeOffer = await this.abundance.observeOffer(need.theirPrivateDid, need.myPrivateSsid)
     await observeOffer.readyPromise
 
-    await this.abundance.getCoreAPI().claim(need.myPrivateSsid, { [this.props.config.SOURCE_ARGUMENT]: this.props.bsn })
-
+    await this.abundance.getCoreAPI().claim(need.myPrivateSsid, {
+      [this.props.config.SOURCE_ARGUMENT]: this.props.bsn,
+      ['ipv8_link']: ipv8TempLink
+    })
 
-    let result = await observeOffer.resultPromise
+    const result = await observeOffer.resultPromise
+    const attestedIpv8Link = result.claim.data
+    const resultLink = result.claim.previous
```

Comment:
I think these claims are in a channel that is used for claims that are meant for multiple parties, so I'm not quite sure if you can rely on the previous claim being the claim that you think it is. I.e. if 2 people request credentials at the same time, this could race, right?

Reply:
I use the channel that is provided/created by the abundance service. Isn't that channel private and only used for the expressed need?

Reply:
We're talking about these claims, correct? https://github.com/discipl/waardepapieren/pull/139/files#diff-4800a0f2d551d933733668eccc7b7e8bR111

Reply:
Yes and now I see what you mean. That is indeed a shared shared thus the `previous` claim is not reliable in this case

Reply:
I've made some changes on this part of the code. Instead of using `previous` I now bundle both claim links into an object which I claim with the nlxIdentity and send it as offer. I think this solves the `previous` problem and the `allow` makes sure the data is only visible for the right parties.